### ISSUE TITLE
[Global - Third party libraries] Make compilation fail if 1/3 party libraries are not installed for external projects

### ISSUE THREAD

**Issue by VincentRouvreau** [GUDHI/gudhi-devel#10]
Compilation shall fail if 1/3 party libraries are not installed for external projects, modulo gudhi_patches for CGAL.

**Comment by VincentRouvreau** [comment]
```cpp
#include <CGAL/version.h>

#if CGAL_VERSION_NR < 1041101000
  // Make compilation fail - required for external projects - https://gitlab.inria.fr/GUDHI/gudhi-devel/issues/10
  static_assert(false);
#endif
```

**Comment by VincentRouvreau** [comment]
Another solution :
```c++
   #if CGAL_VERSION_NR < 1041101000
   // Make compilation fail - required for external projects - https://gitlab.inria.fr/GUDHI/gudhi-devel/issues/10
   #  error Alpha_complex_3d is only available for CGAL >= 4.11
   #endif
```
